doc/developer_guide/psyir.rst

```diff
@@ -744,6 +744,30 @@ subclass of ``Member``. Subsequent children represent the index expressions
 for the array access. The full API is given in the
 :ref_guide:`ArrayOfStructuresMember section of the reference guide psyclone.psyir.nodes.array_of_structures_member.html`.
 
+Data Type of a Structure Access
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+In order to get the actual data type of a structure reference, PSyclone
+needs to have access to the declaration of all structures involved
+in the access. In case of
```

Comment:
I suggest adding:
"... in the accessor expression. However, these are often DeferredType if the module where they are declared has not been processed. In the case of some domain-API arguments added by PSyclone to a kernel call ..."

Reply:
And update copyright authorship and date

Reply:
That sounds better!